src/GenAPI/Microsoft.DotNet.GenAPI/INamedTypeSymbolExtension.cs

```diff
@@ -159,5 +160,54 @@ public static bool IsBoundGenericType(this INamedTypeSymbol namedType)
 
             return true;
         }
+
+        // Synthesize an internal default constructor for the type with implicit default constructor and the base type without.
+        public static IEnumerable<SyntaxNode> TryGetInternalDefaultConstructor(this INamedTypeSymbol namedType, ISymbolFilter symbolFilter)
+        {
+            if (!namedType.Constructors.Where(symbolFilter.Include).Any() && namedType.BaseType != null)
+            {
+                IEnumerable<IMethodSymbol> baseConstructors = namedType.BaseType.Constructors.Where(symbolFilter.Include);
+                if (baseConstructors.Any() && baseConstructors.All(c => !c.Parameters.IsEmpty))
+                {
+                    SyntaxKind visibility = SyntaxKind.InternalKeyword;
+
+                    if ((symbolFilter is AccessibilitySymbolFilter accessibilityFilter && accessibilityFilter.IncludeInternalSymbols) ||
+                        (symbolFilter is CompositeSymbolFilter compositeSymbolFilter &&
```

Comment:
Maybe it'd be good to extract this check into a function, like `IncludeInternalSymbols(symbolFilter)`?

Reply:
Introduced a lambda function:

```C#
Func<ISymbolFilter, bool> IncludeInternalSymbols = filter =>
    filter is AccessibilitySymbolFilter accessibilityFilter && accessibilityFilter.IncludeInternalSymbols;
```